Fix grammatical errors and problems with deprecation notice and removal notice (#164)
redhat-documentation/supplementary-style-guide: supplementary_style_guide/style_guidelines/specific-doc-types.adoc

[discrete]
===== Deprecation notice
[subs="+quotes"]
----

> In __<product_name> <release>__, __<name_of_capability_or_feature>__ is deprecated and is planned to be removed in the __<deprecation_timeline>__. Red Hat will provide bug fixes and support for this feature during the current release life cycle, but this feature will no longer receive enhancements and will be removed. As an alternative to __<name_of_capability_or_feature>__, you can use __<alternative_capability_or_feature_if_available>__ instead.

Reviewer: "life cycle" should be "lifecycle" one word. I know most were here, but would you mind fixing that as well while you're in here?

Author: Thanks @bergerhoffer. I'll fix that.

----
[NOTE]
====
When citing deprecation timelines, refer to specific releases, such as __the next release__, only if that timeline is known to be accurate. Otherwise, use the phrase __a future release__ because it accounts for the possibility of changes to the planned deprecation timeline.
====

.Example deprecation notice doc text
----
In Red Hat Openstack Platform (RHOSP) 14, the director graphical user interface is deprecated and is planned to be removed in a future release. Red Hat will provide bug fixes and support for this feature during the current release life cycle, but this feature will no longer receive enhancements and will be removed.
----

[discrete]
===== Removal notice
[subs="+quotes"]
----

> In __<product_name> <current_release>__, __<name of capability or feature>__ is removed. Bug fixes and support are provided only through the end of the __<previous_release>__ life cycle. As an alternative to __<name_of_capability_or_feature>__, you can use __<alternative_capability_or_feature_if_available>__ instead. 

Reviewer: Even though it might be grammatically correct, "is removed" doesn't sound right to my ears. Wouldn't "has been removed" be better?

Again, this was there before, but might as well consider now since we're fixing other grammatical issues in there. Appreciate thoughts on this one

Author: Good point, @bergerhoffer. I'll fix that.

[discrete]
===== Deprecation notice
----

> In <product> <release>, <name of capability> is deprecated and will be removed in the <deprecation time frame>. Bug fixes and support are provided through the end of the current release life cycle, but be aware that deprecated items receive no further feature enhancements and will be removed.

Reviewer 2: ```suggestion
In <product> <release>, <name of capability> is deprecated and is planned to be removed in the <deprecation time frame>. Bug fixes and support are provided through the end of the current release life cycle. Deprecated items do not receive feature enhancements.
```
Does this suggestion align with the following example and improve clarity?

Reviewer 2: If adopted, apply changes to the following example, too.

Reviewer 3: `<deprecation time frame>` seems ambiguous to me. In what units is this time frame counted? If it's defined in terms of a future release, then it'd be very helpful to me read just that inside `<...>` so that I don't have to stop and think and ask or do something different because I'm pressed for time and can't think straight.

Reviewer 3: Same as for the removal note:

A while back, I had a conversation with an SME who asked me to insert the Optional sentence as the second sentence:

```suggestion
In <product> <release>, <name of capability> is deprecated and will be removed in the <deprecation time frame>. (Optional: You can use <alternative capability> instead.) Bug fixes and support are provided through the end of the current release life cycle, but be aware that deprecated items receive no further feature enhancements and will be removed.
```
![image](https://user-images.githubusercontent.com/78564304/155340740-54e91ec3-b6aa-4aa4-8d45-1410fdefa2e0.png)

Reviewer 3: I also added underscores to the user-replaced values in my suggestion according to https://redhat-documentation.github.io/supplementary-style-guide/#user-replaced-values

Author: > I also added italics and underscores to the user-replaced values in my suggestion according to https://redhat-documentation.github.io/supplementary-style-guide/#user-replaced-values

The other samples in the SSG don't use italics, so to remain consistent I'm not adding them here

----
.Example deprecation notice doc text
----

> In Red Hat Openstack Platform (RHOSP) 14, the director graphical user interface is deprecated and is planned to be removed in the next release. Bug fixes and support are provided through the current release life cycle, but be aware that deprecated items receive no further enhancements and will be removed.

Reviewer 4: Other then a minor comment , lgtm

> In Red Hat Openstack Platform (RHOSP) 14, the director graphical user interface is deprecated and is planned to be removed in the next release. Bug fixes and support are provided through the current release life cycle, but be aware that deprecated items receive no further enhancements and will be removed.

Author: Changed the wording to avoid future tense entirely

----

[discrete]
===== Removal notice
----

> In <product> <current release>, <name of capability> is removed. Bug fixes and support are provided through the end of the <previous release> life cycle.

Reviewer 3: I added underscores to the user-replaced values in my suggestion according to https://redhat-documentation.github.io/supplementary-style-guide/#user-replaced-values

[discrete]
===== Deprecation notice
----
In <product> <release>, <name of capability or feature> is deprecated and is planned to be removed in the <deprecation time frame>. Bug fixes and support are provided through the end of the current release life cycle, but be aware that deprecated items do not receive further enhancements and will be removed. [Optional, if an alternative is available] Alternatively, you can use <alternative capability or feature> instead.
----

> When citing deprecation time frames, refer to specific releases (such as "the next release") only if that time frame is known to be accurate. Otherwise, refer to "a future release" because it gives flexibility to account for changes to the planned deprecation.

Reviewer 5: IBM SG (pg.231) states:

> Future product releases: Avoid claims that are related to the content or timing of a future product or release.

SO should the line with "a future release" be removed?

> When citing deprecation time frames, refer to specific releases (such as "the next release") only if that time frame is known to be accurate. Otherwise, refer to "a future release" because it gives flexibility to account for changes to the planned deprecation.

Reviewer 6: So the release notes are an exception to talking about the future:
https://redhat-documentation.github.io/supplementary-style-guide/#statements-about-the-future

The double quotation thing just came up in a PR I did and for consistency with this guide, we kept them in so I'd suggest the same here. Probably should think about this across the guide at some point though, good point.

> When citing deprecation time frames, refer to specific releases (such as "the next release") only if that time frame is known to be accurate. Otherwise, refer to "a future release" because it gives flexibility to account for changes to the planned deprecation.

Reviewer 5: OK. Thanks, @ndeevy . Must be a product-specific allowance, because I cannot use such language for Runtimes products. I think we got caught out once or twice with such comments that were never delivered.

I'll mark this comment as resolved.

.Example deprecation notice doc text
----

> In Red Hat Openstack Platform (RHOSP) 14, the director graphical user interface is deprecated and is planned to be removed in the next release. Bug fixes and support are provided through the current release life cycle, but be aware that deprecated items do not receive enhancements and will be removed.

Reviewer 5: Avoid future release clause? See previous comment.

Reviewer 5: Sentence:

> but be aware that deprecated items do not receive enhancements and will be removed.

Has no subject. Red Hat should be the subject here.

Author: > Sentence:
> 
> > but be aware that deprecated items do not receive enhancements and will be removed.
> 
> Has no subject. Red Hat should be the subject here.

The subject (you) is implied because it's an imperative.

Reviewer 5: Hi @bburt-rh .

I disagree. Red Hat (team of engineers, and so on) is the correct subject to use here. The documentation user (usually an external customer) would not control enhancements to deprecated features. 

> but be aware that deprecated items do not receive enhancements and will be removed by Red Hat.

Reviewer 6: @Amrita42 thanks, I'm not sure. (need to study some more about through vs throughout tbh!). And 'items' does seem a bit vague. I suppose it's to be broader than just products. Do you have a good alternative? 'components' maybe? maybe that's too specific though. If we don't have a better option I don't think 'items' causes any problems but I agree it's not great.

Reviewer 4: Thanks @ndeevy I could not think of a replacement for **items**  maybe for now we can keep as is.
 I found this link >> https://wikidiff.com/through/throughout   >> imo we should use **throughout**  because we want to tel users that its during the entire duration of the release lifecycle.

Reviewer 4: @bburt-rh wdyt about the rewrite?

Reviewer 4: Once we have this comment addressed , we are good to go :)

Author: @Amrita42 @ndeevy @dfitzmau 
- I think that 'during' is better than `throughout` or `through`, so I will change it.
- I think `features` is better than `items`, and so I will change it.
- I will change the passive construction per the suggestion of @ndeevy 
- I'm rewriting to use future tense, because we're talking about the future.

Here's the update:

`Red Hat will provide bug fixes and support during the life cycle of the current release. Deprecated items will no longer receive enhancements and will be removed.`

----

[discrete]
===== Removal notice
----

> In <product> <current release>, <name of capability or feature> is removed. Bug fixes and support are provided only through the end of the <previous release> life cycle. [Optional, if an alternative capability or feature is available] Alternatively, you can use <alternative capability or feature> instead. 

Reviewer 5: Place user-replaced values in italics. 

See: https://redhat-documentation.github.io/supplementary-style-guide/#user-replaced-values

Reviewer 5: [Optional, if an alternative capability or feature is available] Alternatively, you can use <alternative capability or feature> instead. 

New paragraph 

> Optional:You can use `<alternative capability or feature>` instead.

Reviewer 6: The `Optional:` is usually for an optional step in a procedure, not sure if we need that here since these notices read like paragraphs, maybe it's fine to keep them that way, wdyt @dfitzmau ?

Author: @ndeevy @dfitzmau I will remove the Optional portion of the example. I think people are smart enough to know that if there's no alternative available, then it can't be listed.
